core/db/migrate/20141101231208_fix_adjustment_order_presence.rb

```diff
@@ -5,7 +5,7 @@ def change
       adjustable = adjustment.adjustable
       if adjustable.is_a? Spree::Order
         adjustment.update_attributes!(order_id: adjustable.id)
-      else
+      elsif adjustable && adjustable.order
         adjustment.update_attributes!(adjustable: adjustable.order)
```

Comment:
I'm right there with you. As I stated in my first comment:

"I'm trying to get an idea of what this migration was supposed to do, and why it is a good/bad idea."

And I'm no closer to finding my answer.... :confused: